system-addon/content-src/components/TopSites/TopSite.jsx

```diff
@@ -180,6 +180,30 @@ TopSite.defaultProps = {
 
 const TopSitePlaceholder = () => <TopSiteLink className="placeholder" />;
 
+const TopSiteList = props => {
+  const topSites = props.TopSites.rows.slice(0, props.TopSitesCount);
+  const topSitesUI = [];
+  for (let i = 0, l = props.TopSitesCount; i < l; i++) {
+    const link = topSites[i];
+    topSitesUI.push(!link ? <TopSitePlaceholder key={i} /> : <TopSite
+      key={link.guid || link.url}
+      dispatch={props.dispatch}
+      link={link}
+      index={i}
+      intl={props.intl}
+      onEdit={props.onEdit}
+      editMode={props.editMode} />);
+  }
+  return (<ul className="top-sites-list">
+    {topSitesUI}
+  </ul>);
+};
+TopSiteList.defaultProps = {
+  editMode: false,
+  onEdit() {}
```

Comment:
Without a `defaultProps` for `TopSiteList`, if there's no `onEdit` provided, it should probably imply `editMode: false`? In particular, `TopSiteList` could not need `editMode` and just take an option `onEdit` to unconditionally pass over to `TopSite` which would be updated to conditionally render `LinkMenu` or the `edit-menu` (without its own `defaultProps`) ?

Reply:
👍 I'll do this in a followup PR.